editor/plugins/tile_set_editor_plugin.cpp

```diff
@@ -2918,7 +2918,12 @@ void TileSetEditor::update_texture_list_icon() {
 	for (int current_idx = 0; current_idx < texture_list->get_item_count(); current_idx++) {
 		RID rid = texture_list->get_item_metadata(current_idx);
 		texture_list->set_item_icon(current_idx, texture_map[rid]);
-		texture_list->set_item_icon_region(current_idx, Rect2(0, 0, 150, 100));
+		Size2 textureSize = texture_map[rid]->get_size();
+
+		if (textureSize.x > 150 || textureSize.y > 100)
```

Comment:
This will still stretch the texture if it's e.g. 50x200 (it will stretch to 150x100).

A better fix would be (removing the `if`):
```
texture_list->set_item_icon_region(current_idx, Rect2(0, 0, MIN(texture_size.x, 150), MIN(texture_size.y, 100)));
```